### PR TITLE
Beginning of SSLContext

### PR DESCRIPTION
**Pull request by minor-jason**
Baby steps. This is the first step toward implementing one of our objects. @d2chau @lwahlmeier @flipontheradio @longcinusa  please check SSLContext.java and javax-net-ssl.cpp

### CODE REVIEW

File: classpath/javax-net-ssl.cpp [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// javax-net-ssl.cpp
+#include "jni.h"
+#include "sslcontext.h"
+
+extern "C" JNIEXPORT SSLCONTEXT JNICALL Java_javax_net_ssl_SSLContext_create(JNIEnv* env, jclass) {
```
Comment (lwahlmeier) [inline classpath/javax-net-ssl.cpp]: Pretty sure this needs to return jlong, and then you need to return the pointer to the ctx, not the actual ctx.

File: classpath/javax-net-ssl.cpp [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+    return create(env);
```
Comment (lwahlmeier) [inline classpath/javax-net-ssl.cpp]: not sure why create is taking in env, its not used by the static.

File: classpath/sslcontext.h [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#include <openssl/ssl.h>
+
+static SSL_CTX* create() {
+    SSL_load_error_strings();
+    SSL_library_init();
+    OpenSSL_add_all_algorithms();
```
Comment (lwahlmeier) [inline classpath/sslcontext.h]: I would make a method call something like SSL_LOAD that is called in the static block in SSLContext.java that does this.  there is no reason to initialize this stuff in c when every SSLContext is made.

**Comment by minor-jason** [comment]
Thanks @lwahlmeier . In your opinion, is this the general direction in which we should be heading?

**Comment by lwahlmeier** [comment]
absolutely.  you are pretty much going to be doing this https://github.com/lwahlmeier/openssl-nosocket-example

The only real difference is you will be passing in and out of java to do it.  Honestly even if you have to add some sideloading of the certs into the sslcontext you should be able to get a simple java app doing what that sample does pretty fast.

**Comment by minor-jason** [comment]
@lwahlmeier @d2chau please review recent commits.

**Pull request merged**